IntelliJ IDEA 2019.2.4 UE 无法识别import

我使用的是IntelliJ IDEA 2019.2.4 UE 版本，mac系统，插件版本为6.6。碰到了你的Issue #47 的问题。
尝试提交https://leetcode.com/problems/two-sum/

```java
package com.leetcode.editor.en;

import static java.util.Arrays.fill;

public class TwoSum_1 {
  public static void main(String[] args) {
       Solution solution = new TwoSum_1().new Solution();
  }
  
  //leetcode submit region begin(Prohibit modification and deletion)
class Solution {
      public int[] twoSum(int[] nums, int target) {
          int[] pre = new int[100005];
          int[] res = new int[2];
          int N = 50000;
          fill(pre, -1);
          int len = nums.length;
          for(int i = 0;i < len; i++) {
              pre[nums[i] + N] = i;
          }
//        int judge = target/2;
          for(int i = 0; i < len; i++) {
              if(nums[i] + N <= target + 2 * N) {
                  if(pre[target + N - nums[i]] != -1 && pre[target + N - nums[i]] != i) {
                      res[0] = i;
                      res[1] = pre[target + N - nums[i]];
                      return res;
                  }
              }
          }
          return res;
      }
}
//leetcode submit region end(Prohibit modification and deletion)

}
```
当我尝试submit时，会报错提示无法找到我引入的fill()方法
```shell
12:11 AM	info
					Compile Error:
					Line 6: error: cannot find symbol
					fill(pre, -1);
					^
					symbol:   method fill(int[],int)
					location: class Solution
					Your input:null
					stdout:
					null
```
查询历史submissions时显示的内容如下
```java
class Solution {
      public int[] twoSum(int[] nums, int target) {
          int[] pre = new int[100005];
          int[] res = new int[2];
          int N = 50000;
          fill(pre, -1);
          int len = nums.length;
          for(int i = 0;i < len; i++) {
              pre[nums[i] + N] = i;
          }
//        int judge = target/2;
          for(int i = 0; i < len; i++) {
              if(nums[i] + N <= target + 2 * N) {
                  if(pre[target + N - nums[i]] != -1 && pre[target + N - nums[i]] != i) {
                      res[0] = i;
                      res[1] = pre[target + N - nums[i]];
                      return res;
                  }
              }
          }
          return res;
      }
}

//total_correct:0
//compile_error:Line 6: error: cannot find symbol

```
看起来是直接删掉了我的import语句




我的codeTemplete配置如下  
```shell
${question.content}

package com.leetcode.editor.en;

public class $!velocityTool.camelCaseName(${question.titleSlug})_${question.frontendQuestionId} {
  public static void main(String[] args) {
       Solution solution = new $!velocityTool.camelCaseName(${question.titleSlug})_${question.frontendQuestionId}().new Solution();
  }
  
  ${question.code}
}
```

--- a/src/main/java/com/shuzijun/leetcode/plugin/language/ScopeFoldingBuilder.java
+++ b/src/main/java/com/shuzijun/leetcode/plugin/language/ScopeFoldingBuilder.java
@@ -1,0 +1,68 @@
+package com.shuzijun.leetcode.plugin.language;
+
+import com.intellij.lang.ASTNode;
+import com.intellij.lang.folding.FoldingBuilderEx;
+import com.intellij.lang.folding.FoldingDescriptor;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.FoldingGroup;
+import com.intellij.openapi.project.DumbAware;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.psi.PsiComment;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.shuzijun.leetcode.plugin.model.Constant;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * @author shuzijun
+ */
+public class ScopeFoldingBuilder extends FoldingBuilderEx implements DumbAware {
+    @Override
+    public @NotNull FoldingDescriptor[] buildFoldRegions(@NotNull PsiElement root, @NotNull Document document, boolean b) {
+        FoldingGroup group = FoldingGroup.newGroup("leetcode editor scope");
+        List<FoldingDescriptor> descriptors = new ArrayList<>();
+        // Get a collection of the literal expressions in the document below root
+        Collection<PsiComment> psiComments =
+                PsiTreeUtil.findChildrenOfType(root, PsiComment.class);
+        // Evaluate the collection
+        for (final PsiComment psiComment : psiComments) {
+            String value = psiComment.getText();
+            if (value != null && (value.contains(Constant.SUBMIT_REGION_BEGIN) || value.contains(Constant.SUBMIT_REGION_END))) {
+                // Add a folding descriptor for the literal expression at this node.
+                descriptors.add(new FoldingDescriptor(psiComment.getNode(),
+                        new TextRange(psiComment.getTextRange().getStartOffset(),
+                                psiComment.getTextRange().getEndOffset()),
+                        group));
+            }
+        }
+        return descriptors.toArray(new FoldingDescriptor[descriptors.size()]);
+    }
+
+
+    @Override
+    public @Nullable String getPlaceholderText(@NotNull ASTNode astNode) {
+        String retTxt = "----";
+        if (astNode.getPsi() instanceof PsiComment) {
+            PsiComment psiComment = (PsiComment) astNode.getPsi();
+            String value = psiComment.getText();
+            if (value != null) {
+                if (value.contains(Constant.SUBMIT_REGION_BEGIN)) {
+                    retTxt = "--BEGIN--";
+                } else if (value.contains(Constant.SUBMIT_REGION_END)) {
+                    retTxt = "--END--";
+                }
+            }
+        }
+        return retTxt;
+    }
+
+    @Override
+    public boolean isCollapsedByDefault(@NotNull ASTNode astNode) {
+        return true;
+    }
+}

--- a/src/main/java/com/shuzijun/leetcode/plugin/utils/FileUtils.java
+++ b/src/main/java/com/shuzijun/leetcode/plugin/utils/FileUtils.java
@@ -21,6 +21,7 @@ import com.shuzijun.leetcode.plugin.setting.ProjectConfig;
 import org.apache.commons.lang.StringUtils;
 
 import java.io.*;
+import java.util.ArrayList;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.concurrent.atomic.AtomicReference;
@@ -123,23 +124,25 @@ public class FileUtils {
             if (StringUtils.isNotBlank(body)) {
 
                 List<String> codeList = new LinkedList<>();
-                int codeBegin = -1;
-                int codeEnd = -1;
-                int lineCount = 0;
+                List<Integer> codeBegins = new ArrayList<>();
+                List<Integer> codeEnds = new ArrayList<>();
+                Integer lineCount = 0;
 
                 String[] lines = body.split("\r\n|\r|\n");
                 for (String line : lines) {
                     if (StringUtils.isNotBlank(line) && trim(line).equals(trim(codeTypeEnum.getComment() + Constant.SUBMIT_REGION_BEGIN))) {
-                        codeBegin = lineCount;
+                        codeBegins.add(lineCount);
                     } else if (StringUtils.isNotBlank(line) && trim(line).equals(trim(codeTypeEnum.getComment() + Constant.SUBMIT_REGION_END))) {
-                        codeEnd = lineCount;
+                        codeEnds.add(lineCount);
                     }
                     codeList.add(line);
                     lineCount++;
                 }
-                if (codeBegin >= 0 && codeEnd > 0 && codeBegin < codeEnd) {
-                    for (int i = codeBegin + 1; i < codeEnd; i++) {
-                        code.append(codeList.get(i)).append("\n");
+                if (codeBegins.size() == codeBegins.size() && codeBegins.size() > 0) {
+                    for (int s = 0; s < codeBegins.size(); s++) {
+                        for (int i = codeBegins.get(s) + 1; i < codeEnds.get(s); i++) {
+                            code.append(codeList.get(i)).append("\n");
+                        }
                     }
                 } else {
                     Boolean isCode = Boolean.FALSE;
@@ -252,16 +255,16 @@ public class FileUtils {
         }
     }
 
-    public static void openFileEditor(File file, Project project){
-        ApplicationManager.getApplication().invokeAndWait(()->{
+    public static void openFileEditor(File file, Project project) {
+        ApplicationManager.getApplication().invokeAndWait(() -> {
             VirtualFile vf = LocalFileSystem.getInstance().refreshAndFindFileByIoFile(file);
             OpenFileDescriptor descriptor = new OpenFileDescriptor(project, vf);
             FileEditorManager.getInstance(project).openTextEditor(descriptor, false);
         });
     }
 
-    public static void openFileEditorAndSaveState(File file, Project project, Question question){
-        ApplicationManager.getApplication().invokeAndWait(()->{
+    public static void openFileEditorAndSaveState(File file, Project project, Question question) {
+        ApplicationManager.getApplication().invokeAndWait(() -> {
             VirtualFile vf = LocalFileSystem.getInstance().refreshAndFindFileByIoFile(file);
             OpenFileDescriptor descriptor = new OpenFileDescriptor(project, vf);
             FileEditorManager.getInstance(project).openTextEditor(descriptor, false);

--- a/src/main/java/com/shuzijun/leetcode/plugin/utils/VelocityTool.java
+++ b/src/main/java/com/shuzijun/leetcode/plugin/utils/VelocityTool.java
@@ -1,5 +1,9 @@
 package com.shuzijun.leetcode.plugin.utils;
 
+import com.shuzijun.leetcode.plugin.model.CodeTypeEnum;
+import com.shuzijun.leetcode.plugin.model.Config;
+import com.shuzijun.leetcode.plugin.model.Constant;
+import com.shuzijun.leetcode.plugin.setting.PersistentConfig;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.lang3.time.DateFormatUtils;
 
@@ -88,4 +92,17 @@ public class VelocityTool extends StringUtils {
         return DateFormatUtils.format(new Date(), format);
     }
 
+    public static String SUBMIT_REGION_BEGIN(){
+        Config config = PersistentConfig.getInstance().getInitConfig();
+        String codeType = config.getCodeType();
+        CodeTypeEnum codeTypeEnum = CodeTypeEnum.getCodeTypeEnum(codeType);
+        return codeTypeEnum.getComment() + Constant.SUBMIT_REGION_BEGIN;
+    }
+
+    public static String SUBMIT_REGION_END(){
+        Config config = PersistentConfig.getInstance().getInitConfig();
+        String codeType = config.getCodeType();
+        CodeTypeEnum codeTypeEnum = CodeTypeEnum.getCodeTypeEnum(codeType);
+        return codeTypeEnum.getComment() + Constant.SUBMIT_REGION_END;
+    }
 }

--- a/src/main/resources/META-INF/plugin.xml
+++ b/src/main/resources/META-INF/plugin.xml
@@ -381,6 +381,21 @@
         <applicationConfigurable groupId="tools" displayName="leetcode plugin" id="leetcode.id"
                                  instance="com.shuzijun.leetcode.plugin.setting.SettingConfigurable"/>
         <statusBarWidgetProvider implementation="com.shuzijun.leetcode.plugin.timer.TimerStatusBarWidgetProvider"/>
+        <lang.foldingBuilder language="JAVA" implementationClass="com.shuzijun.leetcode.plugin.language.ScopeFoldingBuilder"/>
+        <lang.foldingBuilder language="Python" implementationClass="com.shuzijun.leetcode.plugin.language.ScopeFoldingBuilder"/>
+        <lang.foldingBuilder language="ObjectiveC" implementationClass="com.shuzijun.leetcode.plugin.language.ScopeFoldingBuilder"/>
+        <lang.foldingBuilder language="CS" implementationClass="com.shuzijun.leetcode.plugin.language.ScopeFoldingBuilder"/>
+        <lang.foldingBuilder language="ECMAScript 6" implementationClass="com.shuzijun.leetcode.plugin.language.ScopeFoldingBuilder"/>
+        <lang.foldingBuilder language="ruby" implementationClass="com.shuzijun.leetcode.plugin.language.ScopeFoldingBuilder"/>
+        <lang.foldingBuilder language="Swift" implementationClass="com.shuzijun.leetcode.plugin.language.ScopeFoldingBuilder"/>
+        <lang.foldingBuilder language="go" implementationClass="com.shuzijun.leetcode.plugin.language.ScopeFoldingBuilder"/>
+        <lang.foldingBuilder language="Scala" implementationClass="com.shuzijun.leetcode.plugin.language.ScopeFoldingBuilder"/>
+        <lang.foldingBuilder language="kotlin" implementationClass="com.shuzijun.leetcode.plugin.language.ScopeFoldingBuilder"/>
+        <lang.foldingBuilder language="Renderscript" implementationClass="com.shuzijun.leetcode.plugin.language.ScopeFoldingBuilder"/>
+        <lang.foldingBuilder language="PHP" implementationClass="com.shuzijun.leetcode.plugin.language.ScopeFoldingBuilder"/>
+        <lang.foldingBuilder language="Shell Script" implementationClass="com.shuzijun.leetcode.plugin.language.ScopeFoldingBuilder"/>
+        <lang.foldingBuilder language="GenericSQL" implementationClass="com.shuzijun.leetcode.plugin.language.ScopeFoldingBuilder"/>
+        <lang.foldingBuilder language="TypeScript" implementationClass="com.shuzijun.leetcode.plugin.language.ScopeFoldingBuilder"/>
     </extensions>
 
     <actions>

